.github/workflows/terraform-tests.yml

```
@@ -0,0 +1,102 @@
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+name: Terraform Tests
+jobs:
+
+  prepare_list:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(python tests/terraform/get-tf-services.py --names)"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  test_service:
+    needs: prepare_list
+    strategy:
+      fail-fast: false
+      matrix:
+        service: ${{ fromJson(needs.prepare_list.outputs.matrix) }}
+        go-version: [1.18.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      DNS_ADDRESS: 0
+      AWS_DEFAULT_REGION: us-east-2
+      AWS_ALTERNATE_REGION: eu-west-1
+
+    steps:
+    - uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Checkout Localstack
+      uses: actions/checkout@v3
+
+    - name: Checkout Terraform AWS Provider
+      uses: actions/checkout@v3
+      with:
+        repository: hashicorp/terraform-provider-aws
+        path: './terraform-provider-aws'
+
+    - name: Set up Python 3.10.5
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10.5'
+
+    - name: Caching Python .venv
+      id: python-cache
+      uses: actions/cache@v2
+      with:
+        path: ./.venv/
+        key: ${{ runner.os }}-venv-${{ hashFiles('**/setup.cfg') }}
+        restore-keys: |
+          ${{ runner.os }}-venv-
+
+    - name: Install system dependencies
+      run: |
+        sudo apt update
+        sudo apt install libsasl2-dev -y
+        pip install --upgrade pip
+        pip install pyyaml
+
+    - name: Get list of tests for this service
+      id: get-list
+      run: echo "::set-output name=testlist::$(python tests/terraform/get-tf-tests.py ${{ matrix.service }})"
+
+    - name: Check Test Lists
+      if: ${{ steps.get-list.outputs.testlist == '' }}
+      run: exit 1
+
+    - name: Installing Localstack Dependancies Start Localstack
+      run: |
+        make install
+        make start &
+
+    - name: Patching Terraform Provider
```

```suggestion
    - name: Patch Terraform Provider
```

Some others could be adjusted as well, so they're all using the same form (e.g. `Caching` => `Cache`)

---

Updated Caching -> Cache, Patching -> Patch